### PR TITLE
[KYUUBI #5065][FOLLOWUP] Graceful close the process when launch engine timeout

### PR DESCRIPTION
**Pull request by ASiegeLion**
### _Why are the changes needed?_
#5065

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by pan3793** [comment]
The diff seems not correct, please rebase your patch against upstream master branch

**Comment by ASiegeLion** [comment]
Why does logCaptureThread not need to be interrupted?

**Comment by ASiegeLion** [comment]
@pan3793 hi ,  the pr need to be modified?

File: docs/deployment/settings.md [hunk 1]
```diff
@@ -419,7 +419,7 @@ You can configure the Kyuubi properties in `$KYUUBI_HOME/conf/kyuubi-defaults.co
...
-| kyuubi.session.engine.startup.destroy.timeout        | PT5S                    | Engine startup process destroy wait time, if the process does not stop after this time, force destroy instead. This configuration only takes effect when `kyuubi.session.engine.startup.waitCompletion=false`.                                                                                                                                                                                                           | duration | 1.8.0 |
+| kyuubi.session.engine.startup.destroy.timeout        | PT5S                    | Engine startup process destroy wait time, if the process does not stop after this time, force destroy instead. This configuration only takes effect when `kyuubi.session.engine.startup.waitCompletion=false`.                                                                                                                                                                                                           | duration | 1.7.2 |
```
Comment (pan3793) [inline docs/deployment/settings.md:422]:
```suggestion
| kyuubi.session.engine.startup.destroy.timeout        | PT5S                    | Engine startup process destroy wait time, if the process does not stop after this time, force destroy instead. This configuration only takes effect when `kyuubi.session.engine.startup.waitCompletion=false`.                                                                                                                                                                                                           | duration | 1.8.0 |
```

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 2]
```diff
@@ -1518,7 +1518,7 @@ object KyuubiConf {
       .doc("Engine startup process destroy wait time, if the process does not " +
         "stop after this time, force destroy instead. This configuration only " +
         s"takes effect when `${SESSION_ENGINE_STARTUP_WAIT_COMPLETION.key}=false`.")
-      .version("1.8.0")
+      .version("1.7.2")
```
Comment (pan3793) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:1521]:
```suggestion
      .version("1.8.0")
```

**Review by pan3793 — APPROVED** [review]

**Review by pan3793 — APPROVED** [review]

**Review by cxzl25 — APPROVED** [review]

**Comment by cxzl25** [comment]
https://github.com/apache/kyuubi/blob/998b5d53493af836bd6d2e06e82842491935aea4/kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchJobSubmission.scala#L249

**Comment by ASiegeLion** [comment]
Ok, I extract a public method

**Comment by pan3793** [comment]
@ASiegeLion I made modifications in https://github.com/apache/kyuubi/pull/5097/commits/d50a388d61493e4f2b8f04095d79b29119756394 based on your change, would you like to take a look? please let me know if you have any concerns.

**Comment by ASiegeLion** [comment]
Ok, submit it again according to your changes

**Comment by ASiegeLion** [comment]
ok

**Comment by pan3793** [comment]
Thanks, merged to master
